### DISCUSSION TITLE
Как запустить проект

### DISCUSSION THREAD

**Discussion by kxknet** [usix79/openquiz#2] [General]
Собираю докер файл из корня на серваке, а он мне говорит:
Step 2/5 : COPY /deploy /
COPY failed: stat /var/lib/docker/tmp/docker-builder140608232/deploy: no such file or directory

**Comment by usix79** [comment]
чтобы собрать докер-контейнер нужно

1. установить (.net sdk)[https://dotnet.microsoft.com/download] (3.1 или 5.0)
2. находясь в корне репозитория выполнить:
2.1 dotnet tool restore
2.2 dotnet fake build -t docker

**Comment by kxknet** [comment]
После последней команды получаю
   Run
Target "doсker" is not defined. Existing targets:
  - Clean
  - InstallClient
  - Build
  - Run
  - PTests
  - Docker
Script reported an error:
-> Target "doсker" is not defined.

**Reply by usix79** [reply]
похоже где-то кириллица в команде. вот эта точно работает
dotnet fake build -t docker

**Comment by kxknet** [comment]
Теперь такое
Build Time Report
---------------------------------------------------------------------
Target          Duration
------          --------
Clean           00:00:00.0030372
InstallClient   00:00:00.0079835   (npm was not found in path. Please install it and make sure it's available from your path. See https://safe-stack.github.io/docs/quickstart/#install-pre-requisites for more info)
Build           00:00:00           (skipped)
Docker          00:00:00           (skipped)
Total:          00:00:00.1213029
Status:         Failure
---------------------------------------------------------------------
Script reported an error:
-> BuildFailedException: Target 'InstallClient' failed.
-> One or more errors occurred. (npm was not found in path. Please install it and make sure it's available from your path. See https://safe-stack.github.io/docs/quickstart/#install-pre-requisites for more info)
-> npm was not found in path. Please install it and make sure it's available from your path. See https://safe-stack.github.io/docs/quickstart/#install-pre-requisites for more info
Hint: To further diagnose the problem you can run fake in verbose mode `fake -v run ...` or set the 'FAKE_DETAILED_ERRORS' environment variable to 'true'
Hint: Detected paket version '5.241.2' in your paket.dependencies file bootstrapper arguments, consider locking the version to '5.249.0' in your dependencies file (/home/dock/openquiz/paket.dependencies).
Read https://github.com/fsharp/FAKE/issues/2193 for details.

**Reply by usix79** [reply]
Точно, для фронтов еще нужна nodejs

Install pre-requisites
You'll need to install the following pre-requisites in order to build SAFE applications

The .NET Core SDK 3.1 https://dotnet.microsoft.com/download/dotnet-core/
node.js (>= 8.0) https://nodejs.org/
npm https://www.npmjs.com/

**Comment by usix79** [comment]
После билда необходимо
1. Создать таблицы DynamoDB
2. Создать S3
3. Cognito
4. GraphQL в AppSync и api-key
5. Все настройки прописать в SystemManager

это на вскидку, то, что помню
